src/47380.c

```diff
@@ -1,14 +1,114 @@
 #include "common.h"
+#include "ld_addrs.h"
+
+void func_8009D21C(s32 arg0, s32* arg1);
+
+typedef struct PhotoData {
+    u8 pad[0x3A0];
+} PhotoData; // Size: 0x3A0
 
-extern s32 D_800AC0F0;
 extern s32 D_800AE27C;
 extern s32 D_800AE280;
-extern s32 D_800B0598;
+extern PhotoData D_800B0598[60]; // Size: 0xD980 - All photos taken in a level
 extern u8 D_800BDF1E;
+extern u8 D_800BDF1C;
+extern u8 D_800BDF1D;
+extern s32 D_800BDF20[3];
+
+// rodata
+extern s32 D_800AC0F0;
+extern char* D_800AE284; // Pokedex entries
+extern s8 D_800AE4E4[];
+// s8 D_800AE4E4[] = {
+//     0,  -1, -1, 1,  2,  3,  4,  -1, -1, -1, 5,  6,  -1, 7,  -1, 8,  -1, -1, -1, -1, -1, -1, -1, -1, 9,  -1,
+//     10, 11, -1, -1, -1, -1, -1, -1, -1, -1, 12, -1, 13, -1, 14, -1, -1, -1, 15, -1, -1, -1, -1, 16, 17, 18,
+//     -1, 19, -1, 20, -1, 21, 22, 23, -1, -1, -1, -1, -1, -1, -1, -1, -1, 24, 25, -1, -1, 26, 27, -1, -1, 28,
+//     29, 30, 31, 32, -1, 33, -1, -1, -1, 34, 35, 36, 37, -1, 38, -1, -1, -1, -1, -1, -1, -1, 39, -1, -1, -1,
+//     -1, -1, -1, -1, 40, -1, -1, -1, 41, -1, 42, -1, -1, 43, -1, 44, 45, -1, 46, 47, 48, 49, -1, -1, 50, 51,
+//     52, 53, 54, -1, -1, -1, 55, -1, -1, -1, -1, -1, 56, 57, 58, 59, 60, -1, 61, -1, 62, 0,  0,  0,  0,  0,
+// };
+extern char D_800AFE80[4];  // = "？";
+extern char D_800AFE84[16]; // = "Ｓｉｇｎ？";
+extern char D_800AFE90[13]; // = "Ｓｉｇｎ";
+extern char D_800AFE9C[4];  // = "Ｓｉｇｎ？";
+
+#ifdef RODATA_MIGRATED
+char* getPokemonName(s32 pkmnID) {
+    if (pkmnID > 0 && pkmnID <= AnimalID_MAX) {
+        return (&D_800AE284)[pkmnID];
+    }
+    if (pkmnID == 0x3EC || pkmnID == 0x3F2 || pkmnID == 0x3FA || pkmnID == 0x3FE || pkmnID == 0x404 ||
+        pkmnID == 0x40B) {
+        if (func_800BFCA0_5CB40(5) == 0) {
+            return "？";
+        }
+    }
+    if (pkmnID == 0x3EC || pkmnID == 0x3F2 || pkmnID == 0x3FA || pkmnID == 0x3FE || pkmnID == 0x404 ||
+        pkmnID == 0x40B) {
+        if (func_800BF3D4_5C274(pkmnID) == 0) {
+            return "Ｓｉｇｎ？";
+        }
+    }
+    switch (pkmnID) {
+        case 0x25B:
+            return (&D_800AE284)[AnimalID_SHELLDER];
+        case 0x3EC:
+        case 0x3F2:
+        case 0x3FA:
+        case 0x3FE:
+        case 0x404:
+        case 0x40B:
+            return "Ｓｉｇｎ";
+        case 0x1F4:
+        case 0x258:
+        case 0x259:
+        case 0x25A:
+            return "Ｓｉｇｎ？";
```

Comment:
This string is wrong
```suggestion
            return "？";
```